src/eval/mod.rs

```diff
@@ -1184,6 +1184,51 @@ impl Eval for ast::Closure {
     }
 }
 
+impl ast::Pattern {
+    // Destruct the given value into the pattern.
+    pub fn define(&self, vm: &mut Vm, value: Value) -> SourceResult<Value> {
+        match self.bindings() {
+            ast::PatternKind::Ident(ident) => {
+                vm.define(ident, value);
+                Ok(Value::None)
+            }
+            ast::PatternKind::Destructure(pattern) => {
+                let Ok(value) = value.clone().cast::<Array>() else {
```

Comment:
is this clone necessary?

Reply:
It might not be, but this is the only way I got it to compile. In the else branch `value.type_name()` is called and otherwise it complains that a moved value is borrowed.